MAUI/DataGrid/row-drag-and-drop.md

```diff
@@ -0,0 +1,402 @@
+
+N> Now you can drag and drop the grid rows using your mouse in the UWP (Desktop) platform.
+
+## Row drag-and-drop template
+
+The data grid allows you to load desired content when performing row drag-and-drop operations using the `SfDataGrid.RowDragDropTemplate`. The template can be defined either in code or XAML.  
+
+## Default template
+
+The default template will be used if no other template is explicitly assigned for row drag-and-drop operations.
+
+<img alt="DataGrid row drag and drop" src="Images\row-drag-and-drop\Dragging.gif" width="404"/>
+
+## Customizing row drag-and-drop template
+
+Any type of custom view can be loaded inside the `SfDataGrid.RowDragDropTemplate`.
+
+Please refer to the following code example that demonstrates how to load a view in a template:
+
+{% tabs %}
+{% highlight xaml %}
+
+        <syncfusion:SfDataGrid.RowDragDropTemplate>
+        <DataTemplate>
+            <Grid BackgroundColor="LightBlue">
+                <Label Text="Drag View"       
+                       TextColor="Black"           
+                       HorizontalTextAlignment="Center" 
+                       VerticalTextAlignment="Center"/>
+            </Grid>
+        </DataTemplate>
+    </syncfusion:SfDataGrid.RowDragDropTemplate>
+</syncfusion:SfDataGrid>
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Default drag and drop" src="Images\row-drag-and-drop\LabelTemp.gif" />
+
+N> Currently, the row drag-and-drop features cannot be used if different row sets with varying heights are specified using the `QueryRowHeight` event.
+
+## Events in row drag-and-drop
+
+The `QueryRowDragging` event is triggered when a row begins to be dragged and continues to be triggered until the dragging ends. By handling the `SfDataGrid.QueryRowDragging` event, you can also cancel the dragging of a specific row.
+
+The `QueryRowDragging` event provides the following properties in the `QueryRowDraggingEventArgs`:
+
+ * `From`: Returns the index of the row that is currently being dragged.
+ * `To`: Returns the index of the row where you attempt to drop the dragged item.
+ * `Position`: Returns the current x and y coordinates of the RowDragView.
+ * `DraggingAction`: Returns the details of row dragging as `DataGridDragAction`.
+ * `RowData`: Returns the underlying data associated with the row that was dragged.
+ * `CurrentRowData`: Returns the row data corresponding to the current position of the row drag view.
+ * `CanAutoScroll`: Returns whether auto-scrolling should occur when the row drag view reaches the top or bottom of the `SfDataGrid`.
+ * `Cancel`: Returns a Boolean value that indicates whether to cancel the event.
```

Comment:
The `QueryRowDragging` event is triggered when a row begins to be dragged and continues to fire until the dragging ends. By handling the `SfDataGrid.QueryRowDragging` event, you can also cancel the dragging of a specific row.

The `QueryRowDragging` event provides the following properties through the `QueryRowDraggingEventArgs`:

 * `From`: Returns the index of the row that is currently being dragged.
 * `To`: Returns the index of the row where you are attempting to drop the dragged item.
 * `Position`: Returns the current x and y coordinates of the `RowDragView`.
 * `DraggingAction`: Returns details of the row dragging as a `DataGridDragAction`.
 * `RowData`: Returns the underlying data associated with the dragged row.
 * `CurrentRowData`: Returns the data corresponding to the current position of the `RowDragView`.
 * `CanAutoScroll`: Returns a value indicating whether auto-scrolling should occur when the `RowDragView` reaches the top or bottom of the `SfDataGrid`.
 * `Cancel`: Returns a Boolean value that indicates whether the event should be canceled.

Reply:
Updated